pkg/controller/vitesscluster/reconcile_topo.go

```diff
@@ -85,43 +85,58 @@ func (r *ReconcileVitessCluster) reconcileCellTopology(ctx context.Context, vt *
 		desiredCells[cell.Name] = cell
 	}
 
-	// We need to add an alias for all the cells in each region so that vtgate
-	// knows that it can route traffic between them.
-	// Currently, only one region is supported so we allow routing anywhere.
-	//
-	// We also need to create the aliases before we create the cells because we
-	// don't want any vtgates to start after the cells are created but before
-	// the alias exists.
-	desiredCellsAliases := buildCellsAliases(desiredCells)
-	currentCellsAliases, err := ts.GetCellsAliases(ctx, true)
+	if *vt.Spec.TopologyReconciliation.RegisterCellsAliases {
+		err := r.registerCellsAliases(ctx, vt, ts, desiredCells)
+		if err != nil {
+			return resultBuilder.Error(err)
+		}
+	}
+
+	if *vt.Spec.TopologyReconciliation.RegisterCells {
+		result, err := r.registerCells(ctx, vt, ts, globalTopoImpl, desiredCells)
+		resultBuilder.Merge(result, err)
+	}
+
+	if *vt.Spec.TopologyReconciliation.PruneCells {
+		result, err := r.pruneCells(ctx, vt, ts, desiredCells)
+		if err != nil {
+			return resultBuilder.RequeueAfter(topoRequeueDelay)
+		}
+		resultBuilder.Merge(result, err)
+	}
+
+	return resultBuilder.Result()
+}
+
+func (r *ReconcileVitessCluster) pruneCells(ctx context.Context, vt *planetscalev2.VitessCluster, ts *topo.Server, desiredCells map[string]*planetscalev2.VitessCellTemplate) (reconcile.Result, error) {
+	resultBuilder := &results.Builder{}
+
+	// Get list of cells in topo.
+	cellNames, err := ts.GetCellInfoNames(ctx)
 	if err != nil {
-		r.recorder.Eventf(vt, corev1.EventTypeWarning, "TopoCellAlias",
-			"Failed to get current cell aliases: %v", err)
+		r.recorder.Eventf(vt, corev1.EventTypeWarning, "TopoListFailed", "failed to list cells in topology: %v", err)
 		return resultBuilder.Error(err)
```

Comment:
This used to be a return RequeueAfter so we should keep it that way.

Reply:
I got you fam https://github.com/planetscale/vitess-operator/pull/26/commits/89d4572f7eda837d744c2fdb304573b5009f849d.